web/src/user_profile.js

```diff
@@ -307,10 +308,14 @@ export function get_custom_profile_field_data(user, field, field_types) {
 }
 
 export function hide_user_profile() {
-    user_streams_list_widget = undefined;
     modals.close_if_open("user-profile-modal");
 }
 
+function on_user_profile_hide() {
+    user_streams_list_widget = undefined;
+    user_profile_subscribe_widget = undefined;
+}
```

Comment:
I think it's time to chat a bit about writing good commit series, since it's an important skill that every new contributor needs to focus on.

This PR should be two commits; the first making this fix to the cleanup process (with a clear commit message explaining why it's an improvement), and the second fixing the main bug for the issue here. And the second one should explain why it's the right solution -- to be honest I am a bit suspicious that it isn't; the code path for other panels clearly expects us to just fully rebuild the panel when visiting it again, and your PR description doesn't explain concretely why doing so causes an error or what options get out of sync.

Reply:
@timabbott , reason behind the other tabs not being affected by fully rebuilding them when re-visited is that they don't have options to make edits to the Profile from the User Profile Modal itself (they lead to opening of some other modal). However, only in case of the Streams Tab, you are editing it from the Modal itself.

Hence, rebuilding the Streams panel is going to cause the issue since the old options values are made null when you try to access them. Hence, the error.

Reply:
If you were referring to other modals by other panels, they are preserving the state when you select an option and come back to the same tab and use the previous state to build it again (because the save/discard option is) still visible.

Reply:
Plus, I feel like it makes more sense if once we select some option in dropdown and move to next tab, and come back... rather than re building the entire panel we should be storing the state and setting it up using the current object since the user doesn't want to lose a choice he filled once and accidentally clicked on other panel.